frb_codegen/src/parser/mod.rs

```diff
@@ -16,9 +18,73 @@ use crate::method_utils::FunctionName;
 use crate::parser::ty::TypeParser;
 use crate::source_graph::Crate;
 
+use self::ty::convert_ident_str;
+
 const STREAM_SINK_IDENT: &str = "StreamSink";
 const RESULT_IDENT: &str = "Result";
 
+pub(crate) fn get_toposort(src: HashMap<String, Type>) -> HashMap<String, Type> {
```

Comment:
given that this function is very isolated, it would be great to add a bit of [unit test](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html) to this one function. it will also make readers clear about sample inputs and outputs of the function